edgeless_node/src/resources/dda/mod.rs

```diff
@@ -80,8 +79,11 @@ impl DDAResource {
         let dda_sub_array = match dcs {
             Ok(dda_array) => dda_array,
             Err(err) => {
-                eprintln!("Error parsing input dda_com_subscription_mapping JSON: {}", err);
-                process::exit(1);
+                log::error!("Error parsing input dda_com_subscription_mapping JSON: {}", err);
+                panic!("Error parsing input dda_com_subscription_mapping JSON: {}", err);
```

Comment:
That will still kill the node.

Re the invariant discussion, this should not leave the node in an undesired state as that only happens at the time of the resource instant start, which means you can catch it (in the API below or by waiting for this to complete with a Result), and signal to the orchestrator that the config was invalid.

We don't have a way to catch dying functions/resource instances after their start.

What you're doing right now is killing the node for an HTTP 400 error in one of the resources.

Reply:
will go for an update on the exit / panic part as discussed in Chanaia in a separate pull request